sleepy/lexer.py

```diff
@@ -19,8 +19,8 @@ def __init__(self, token_names, token_regex_table):
     :param list[str] token_regex_table: corresponding regex's, not recognizing the empty word.
     """
     assert len(token_names) == len(token_regex_table)
-    self.token_names = token_names
-    self.token_regex_table = token_regex_table
+    self.token_names: List[Optional[str]] = token_names
```

Comment:
Pycharm should just copy the type info from the arguments, so this should be unnecessary